### DISCUSSION TITLE
[LSP] - NetBeans should kill nodejs subprocess after closing TypeScript file or project

### DISCUSSION THREAD

**Discussion by Chris2011** [apache/netbeans#5325] [Ideas]
I think this is a general LSP thing, not only for TypeScript files but I just see this in my cases.

I'm not deep enough into this whole topic, but when I open my NetBaens with 3 HTML projects where 2 projects have pieces of TypeScript files inside, NetBeans has a lot nodejs subprcess of round about 38-44 depends on... (After 5 - 10 min testing, opening files, etc.)? I just know that the lsp server for TS started when there is a TypeScript lsp call like opening a file, select the file -> open navigator -> show symbols, etc. But when I close my projects, it never kills those processes. Maybe some but not all. I remember that this was never intended to do.

I tried it with VS Code, problematic is here, the sub processes are most of the time called "vs code", so I can't see enough to figure out what it is (sometimes plugins). I used the process explorer from sysinternals to see through it.

I had a look into WebStorm and just saw a bit how they handled. There is just one WebStorm.exe process, nothing more (maybe everything is hidden) and separated node.js process, but independent from WebStorm. Sometimes when I open vue or ts files, some new node.js processes came up and sometimes they shutdown when I close some files.

In general:
* Open some html5 or Node projects where you have typescript stuff inside. 3-4 are enough and couble of node modules.
* Open the taskmanager
* Navigate to NetBeans
* Expand NetBeans and see a lot of Nodejs process.

I figured out, that somehow 3 nodejs process and 3 console hosts subprocesses started and maybe needed for one project. So it could be that we have 9-12 nodejs processes for 3-4 projects. That doesn't sound much, but first they never got killed after closing a project and also sometimes we start more node subprocesses. For example, when you open node_modules -> open any dependency folder -> click (single/double click) on any *.d.ts file. For some files, we create some more processes. Doesn't matter we opened the folder as project or not due to the package.json file.

So long story short, is so much sub process needed? Can't we kill them? 3 - 4 HTML5 projects are enough to load NetBeans to 1 - 2 GB.

**Comment by Chris2011** [comment]
This happens on a Windows 11 Machine (64bit) with latest NetBeans but also with 15, 14. OpenJDK 14  I just want to discuss this first, before I created a feature request.
